src/Microsoft.Sbom.Targets/GenerateSbomTask.cs

```diff
@@ -121,6 +122,22 @@ public override bool Execute()
     {
         try
         {
+            // Check if all arguments involving paths are rooted
+            if (!Path.IsPathRooted(this.BuildDropPath) ||
+                (!string.IsNullOrEmpty(this.BuildComponentPath) && !Path.IsPathRooted(this.BuildComponentPath)) ||
+                (!string.IsNullOrEmpty(this.ManifestDirPath) && !Path.IsPathRooted(this.ManifestDirPath)) ||
+                (!string.IsNullOrEmpty(this.ExternalDocumentListFile) && !Path.IsPathRooted(this.ExternalDocumentListFile)))
+            {
+                Log.LogError("SBOM generation failed: Unrooted path detected. Please ensure you are specifying full paths for " +
+                    "BuildDropPath, BuildComponentPath, ManifestDirPath, and ExternalDocumentListFile arguments.");
+                return false;
+            }
+
+            if (!(string.IsNullOrEmpty(NamespaceUriUniquePart) || Guid.TryParse(NamespaceUriUniquePart, out var guidResult))) {
```

Comment:
For more readability, I would write it like this:
```suggestion
            if (!string.IsNullOrEmpty(NamespaceUriUniquePart) && !Guid.TryParse(NamespaceUriUniquePart, out var guidResult)) {
```

Reply:
Updated!